src/server/zset_family.cc

```diff
@@ -870,61 +898,187 @@ OpResult<AddResult> OpAdd(const OpArgs& op_args, const ZParams& zparams, string_
   return aresult;
 }
 
-struct StoreArgs {
+struct SetOpArgs {
   AggType agg_type = AggType::SUM;
   unsigned num_keys;
   vector<double> weights;
+  bool with_scores = false;
 };
 
-OpResult<StoreArgs> ParseStoreArgs(CmdArgList args) {
-  string_view num_str = ArgS(args, 2);
-  StoreArgs store_args;
+OpResult<void> FillAggType(string_view agg, SetOpArgs* op_args) {
+  if (agg == "SUM") {
+    op_args->agg_type = AggType::SUM;
+  } else if (agg == "MIN") {
+    op_args->agg_type = AggType::MIN;
+  } else if (agg == "MAX") {
+    op_args->agg_type = AggType::MAX;
+  } else {
+    return OpStatus::SYNTAX_ERR;
+  }
+  return OpStatus::OK;
+}
+
+OpResult<unsigned> ParseAggregate(CmdArgList args, bool store, SetOpArgs* op_args) {
+  if (store && args.size() != 2) {
```

Comment:
This check assumes also that aggregate is the last one in optional args and will fail on
zunionstore a 2 z1 z2 aggregate max WEIGHTS 1 0
although its a valid command